### DISCUSSION TITLE
Telnet REPL

### DISCUSSION THREAD

**Discussion by moferreira** [micropython/micropython#16092] [General]
Is is there a way to do an analogous telnet to the device's  REPL?
Why this question? Because I need to communicate with the device when it is on AP mode so, it has no Internet access. WebREPL works but I need to have the Python application to run it as, when connected to the device in AP mode, I won't have Internet access. However, I want to be able to send commands to it let's say, using my Android device.

**Comment by robert-hh** [comment]
There was/is a Python telnet server script https://github.com/cpopp/MicroTelnetServer.
It used to work, but was not updated since 5 years. So it may need some rework. Which port do you use?

**Reply by moferreira** [reply]
It could be any port. Anyway, I will take a look on the MicroTelnetServer you mentioned.

**Reply by moferreira** [reply]
Tested. Not working... By the way, I am on an ESP32 C6, with the preview firmware.

**Reply by robert-hh** [reply]
I just tested the enclosed version on a ESP32 generic and a ESP32C6, and it worked. It differs slightly from the version on the web.

[utelnetserver.zip](https://github.com/user-attachments/files/17535192/utelnetserver.zip)

Edit: I see that it has a hiccup when the client disconnects by not noticing that fact properly. But it still continues to run.

**Reply by moferreira** [reply]
Solved. Many thanks!
